File: RecyclerViewExercise/RecyclerViewExercise1/app/src/main/java/ai/tomorrow/recyclerviewexercise1/MainActivity.java
package ai.tomorrow.recyclerviewexercise1;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);

        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        List<String> fakeData = new ArrayList<>();
        fakeData.add("fake data 1");
        fakeData.add("fake data 2");
        fakeData.add("fake data 3");
        fakeData.add("fake data 4");
        fakeData.add("fake data 5");
        fakeData.add("fake data 6");
        fakeData.add("fake data 7");

        adapter = new MyAdapter(fakeData);
        recyclerView.setAdapter(adapter);
    }
}

File: RecyclerViewExercise/RecyclerViewExercise2_kotlin/app/src/main/java/ai/tomorrow/recyclerviewexercise2_kotlin/MainActivity.kt
package ai.tomorrow.recyclerviewexercise2_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val TAG: String = "MainActivity"
    private lateinit var mRecyclerView: RecyclerView
    private lateinit var mLayoutManager: LinearLayoutManager
    private lateinit var mAdapter: MyAdapter


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val fakeData = ArrayList<String>()
        fakeData.add("test1")
        fakeData.add("test2")
        fakeData.add("test3")
        fakeData.add("test4")
        fakeData.add("test5")
        fakeData.add("test6")


        mLayoutManager = LinearLayoutManager(this)
        mAdapter = MyAdapter(fakeData)

        mRecyclerView = findViewById<RecyclerView>(R.id.recyclerView).apply{
            setHasFixedSize(true)
            layoutManager = LinearLayoutManager(context)
            adapter = mAdapter
        }
    }
}
